share/spack/setup-env.sh

```diff
@@ -114,12 +114,16 @@ spack() {
                 command spack env -h
             else
                 case $_sp_arg in
+                    # get --sh, --csh, --help, or -h arguments
+                    # space is important for -h case to differentiate between `-h`
+                    # argument and environments with "-h" in the name
                     activate)
-                        _a="$@"
+                        _a=" $@"
```

Comment:
A couple notes:

* There are two instances of spaces here and it's probably worth explaining both and providing an example of each in the comments
  * `_a` has a prepended space because the regexes start with a space and `-h` may be the only argument (`$@` == "-h")
  * The regular expression has a prepended space because we want to differentiate between an argument that starts with `-h` from an argument that includes `-h` with a prefix
* This is also required for the check of `deactivate` below.

While we're here, can you add a link to the documentation that describes this regex syntax? https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html#Shell-Parameter-Expansion is the best one I can find. I waste a bunch of time chasing that down bash regex syntax explanations each time I look at this file.

Reply:
I just stole that comment from the load|unload section below. I'm hesitant to document all of this in like 5 different places...

Reply:
@scheibelp what do you think about my above comment?

Reply:
I think by "that comment" you mean

> get --sh, --csh, --help, or -h arguments

My request doesn't have to do with that comment (which documents the overall reason for the entire code block). I want to document the individual line concerning regex syntax. That only needs to appear in one place (even if the syntax is used in multiple places in the code).

Also, https://github.com/spack/spack/pull/15429/files#r392527024 responds to 

> While we're here, can you add a link to the documentation that describes this regex syntax?

but not 

> This is also required for the check of deactivate below.

(I think this is a bug)

or

> There are two instances of spaces here and it's probably worth explaining both and providing an example of each in the comments

(in this case I'm asking about documenting something that is entirely new: your introduction of a `' '` as a prefix to the regex - that doesn't have to occur multiple times)

So I'm curious if you agree with those two

Reply:
By "that comment" I meant:

> I just stole that comment from the load|unload section below. I'm hesitant to document all of this in like 5 different places...